graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -528,38 +530,73 @@ def _rename_or_suppress_or_ignore_name_and_add_to_record(
             - SchemaNameConflictError if the newly renamed node causes name conflicts with
               existing types, scalars, or builtin types
         """
-        name_string = node.name.value
+        type_name = node.name.value
 
-        if name_string == self.query_type or name_string in self.scalar_types:
+        if (
+            type_name == self.query_type
+            or type_name in self.custom_scalar_names
+            or type_name in builtin_scalar_type_names
+        ):
             return IDLE
 
-        new_name_string = self.renamings.get(name_string, name_string)  # Default use original
-        if new_name_string is None:
+        desired_type_name = self.renamings.get(type_name, type_name)  # Default use original
+        if desired_type_name is None:
             # Suppress the type
             return REMOVE
-        check_type_name_is_valid(new_name_string)
-
+        check_type_name_is_valid(desired_type_name)
+
+        # Renaming conflict arises when two types with different names in the original schema have
+        # the same name in the new schema. There are two ways to produce this conflict:
+        # * when neither type is a custom scalar
+        # * when one type is a custom scalar and the other is not
```

Comment:
nit: i would tab and/or replace the `*` with numbers. visually the `# *` just looks kind of odd to me.